frontend/src/pages/signup.js

```diff
@@ -26,7 +26,25 @@ function Signup() {
 
     const handleRegister = async (e) => {
         e.preventDefault();
-        const user = { username, email, password, password2, role };
+
+        if (password !== password2) {
+            toast.error('Passwords do not match', {
+                position: 'top-center',
+                autoClose: 3000,
+                theme: 'dark',
```

Comment:
there is actually a toast style under styles/toastStyles.js instead of repeating this every time